saleor/discount/models.py

```diff
@@ -225,24 +221,50 @@ class Meta:
         )
 
     def __repr__(self):
-        return "Sale(name=%r, value=%r, type=%s)" % (
-            str(self.name),
-            self.value,
-            self.get_type_display(),
-        )
+        return "Sale(name=%r, type=%s)" % (str(self.name), self.get_type_display(),)
 
     def __str__(self):
         return self.name
 
-    def get_discount(self):
+    def get_discount(self, channel: Channel):
+        sale_channel_listing = self.channel_listing.filter(channel=channel).first()
```

Comment:
Making database queries in model functions is a bad idea as it often leads to unwanted duplicated DB queries. We should either refactor this function to move it outside of the model or pass sale_channel_listing as an argument.